cinema/views.py

```diff
@@ -2,0 +1,70 @@
+from typing import Type
+
+from rest_framework import viewsets
+from rest_framework.serializers import Serializer
+
+from cinema.models import (
+    Movie,
+    Genre,
+    Actor,
+    MovieSession,
+    CinemaHall
+)
+from cinema.serializers import (
+    MovieSerializer,
+    GenreSerializer,
+    ActorSerializer,
+    MovieSessionSerializer,
+    CinemaHallSerializer,
+    MovieListSerializer,
+    MovieSessionListSerializer,
+    MovieSessionRetrieveSerializer,
+    MovieRetrieveSerializer
+)
+
+
+class MovieViewSet(viewsets.ModelViewSet):
+    queryset = Movie.objects.all()
+
+    def get_serializer_class(self) -> Type[Serializer]:
+        if self.action == "list":
+            return MovieListSerializer
+        elif self.action == "retrieve":
+            return MovieRetrieveSerializer
+        return MovieSerializer
+
+    def get_queryset(self) -> queryset:
+        if self.action in ("list", "retrieve"):
+            return self.queryset.prefetch_related("genres", "actors")
+        return self.queryset
+
+
+class GenreViewSet(viewsets.ModelViewSet):
+    queryset = Genre.objects.all()
+    serializer_class = GenreSerializer
+
+
+class ActorViewSet(viewsets.ModelViewSet):
+    queryset = Actor.objects.all()
+    serializer_class = ActorSerializer
+
+
+class MovieSessionViewSet(viewsets.ModelViewSet):
+    queryset = MovieSession.objects.all()
+
+    def get_serializer_class(self) -> Type[Serializer]:
+        if self.action == "list":
+            return MovieSessionListSerializer
+        elif self.action == "retrieve":
+            return MovieSessionRetrieveSerializer
+        return MovieSessionSerializer
+
+    def get_queryset(self) -> queryset:
```

Comment:
Similar to `MovieViewSet`, the return type annotation for `get_queryset` should be `QuerySet` instead of `queryset`. Ensure that `QuerySet` is imported from Django's ORM.